woocommerce/woocommerce-android
# Add product path to product leaderboard request when available

Closes: #6688 

Do not merge. This code changes require [these FluxC changes](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2519) to work and to be tested. 

### Description
`/wc-analytics/leaderboards/products` Was added along with individual endpoints for the other leaderboards. Since we only need the product leaderboard in the apps, we should switch to that new endpoint. This will be faster, lessen the load on the origin site, and transfer less data that we don't need.

### Testing instructions
- Open a site that has a WooCommerce plugin version `6.7.0` or higher
- Check that top performers are loaded correctly in MyStore tab
- Do the same for a site that has an older version than `6.7.0`. If you don't have a site with an older version of WC plugin, please ping me and I can invite you to one. 

### Images/gif

https://user-images.githubusercontent.com/2663464/187480051-e9317c5f-f7c4-499c-b0b6-d02b2f817ebd.mp4
